projects/plugins/crm/includes/ZeroBSCRM.FormatHelpers.php

```diff
@@ -850,6 +850,44 @@ function zeroBSCRM_html_quoteStatusLabel($quote=array()){
 
 	}
 
+/**
+ * Return an updated HTML string, replacing quote date placeholders with correct date strings based on site settings.
+ * @param  string  $value         The value of the date variable to be replaced.
+ * @param  string  $key           The key of the date variable to be replaced.
+ * @param  string  $working_html  The HTML string to be updated.
+ * @param  boolean $custom_field Whether the key in question is a custom quote field.
+ *
+ * @return string                The updated HTML string.
+ */
+function jpcrm_process_date_variables( $value, $key, $working_html, $custom_field = false ) {
+	$placeholder_str_start = '##';
+	if ( $custom_field ) {
+		$placeholder_str_start = '##QUOTE-';
+	}
+	$numeric_date_key       = $key;
+	$string_to_numeric_date = wp_date( get_option( 'date_format' ), strtotime( $value ) );
+	$datetime_key           = $key . '_datetime_str';
+	$string_to_datetime     = wp_date( 'd F Y ' . get_option( 'time_format' ), strtotime( $value ) );
+	$date_key               = $key . '_date_str';
+	$string_to_date         = wp_date( 'd F Y', strtotime( $value ) );
+
+	$search_replace_pairs = array(
+		$placeholder_str_start . strtoupper( $numeric_date_key ) . '##' => $string_to_numeric_date,
+		$placeholder_str_start . strtolower( $numeric_date_key ) . '##' => $string_to_numeric_date,
+		$placeholder_str_start . strtolower( $numeric_date_key ) . '##' => $string_to_numeric_date,
```

Comment:
Why is this here once uppercase and once lowercase? Same question with the items below.

Reply:
In this case I'm just copying the catch-all method used in `ZeroBSCRM_get_quote_template` to catch various uses. Without having full knowledge of when different cases could arise, I opted for using the same fallback that exists already.

Reply:
Ah, but looking in the function from which the fallbacks came, there's a subtle difference; the second and third line are different:

```
$working_html = str_replace( '##QUOTE-' . strtolower( $key ) . '##', $v, $working_html );
$working_html = str_replace( '##quote-' . strtolower( $key ) . '##', $v, $working_html );
```

In your version, the second and third lines are identical. To match the source you'd want to do an uppercase and lowercase variant of `$placeholder_str_start`. Or, since these are placeholders that didn't work well anyway, you could just drop the fallbacks and force the use of the uppercase one. :^)

Reply:
That sounds like a much better approach :D